.github/workflows/pull_request.yml

```diff
@@ -41,45 +38,42 @@ jobs:
         run: |
           echo "PROJECT_NAME=$(cargo metadata --no-deps --format-version 1 | jq -r '.packages[0] | [ .name ] | join("")')" >> $GITHUB_ENV
           echo "PROJECT_VERSION=$(cargo metadata --no-deps --format-version 1 | jq -r '.packages[0] | [ .version ] | join("")')" >> $GITHUB_ENV
-        
+
       - name: Update dependencies
-        uses: actions-rs/cargo@v1
-        with:
-          command: update
+        run: cargo update
       - name: Build
-        uses: actions-rs/cargo@v1
-        with:
-          command: build
-
-      # run tests, once normally, once to calculate code coverage
-      - name: Test
-        uses: actions-rs/cargo@v1
-        with:
-          command: test
-          args: --verbose
-      - name: Run tarpaulin to calculate code coverage
-        uses: actions-rs/tarpaulin@v0.1
-        with:
-          version: '0.15.0'
-          args: '-- --test-threads 1'
-          out-type: Html
-      
-      - name: Run clippy
-        uses: actions-rs/clippy-check@v1
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          args: -- -D warnings # escalate warnings s.t. pipeline fails
-      - name: Format
-        run:  cargo fmt --all -- --check
+        run: cargo build
+      - name: Generate docs
+        run: cargo doc

```

Comment:
How about always running with `RUSTDOCFLAGS "-D warnings"` and using `continue-on-error: true` in the pipelines we don't want to fail because of doc issues?
I personally would like it if doc warnings where introduced less often (currently again some on dev).